console/collections/src/merkle_tree/mod.rs

```diff
@@ -257,18 +271,28 @@ impl<E: Environment, LH: LeafHash<Hash = PH::Hash>, PH: PathHash<Hash = Field<E>
         }
         lap!(timer, "Hashed {} padding levels", padding_depth);
 
-        // Update the root hash.
-        self.root = root_hash;
+        // Initialize the Merkle tree.
+        let mut tree = Vec::with_capacity(self.tree.len());
+        // Extend the new Merkle tree with the existing leaf hashes.
+        tree.extend(&self.tree);
```

Comment:
@ljedrz is `.clone()` faster or is the current logic better?

Reply:
There should be little difference performance-wise; a `clone` would be simpler, the capacity might just differ slightly.